src/sagemaker/model.py

```diff
@@ -407,6 +409,14 @@ def _compilation_job_config(
             else input_shape,
             "Framework": framework.upper(),
         }
+
+        if (
+            framework.lower() == "pytorch"
+            and re.match("(?=^ml_)(?!ml_inf)", target_instance_type) is not None
+        ):
+            if framework_version is not None:
+                input_model_config["FrameworkVersion"] = utils.get_short_version(framework_version)
```

Comment:
```suggestion
        if (
            framework.lower() == "pytorch"
            and re.match("(?=^ml_)(?!ml_inf)", target_instance_type) is not None
            and framework_version is not None
        ):
            input_model_config["FrameworkVersion"] = utils.get_short_version(framework_version)
```

Reply:
Thanks for reviewing. I will make the suggested changes.